orttraining/orttraining/training_api/module.cc

```diff
@@ -186,14 +186,24 @@ Module::Module(const std::string& train_model_path_or_bytes,
   }
 }
 
-size_t Module::GetTrainModeOutputCount() const noexcept {
+size_t Module::GetTrainModelOutputCount() const noexcept {
   return train_output_names_.size();
 }
 
-size_t Module::GetEvalModeOutputCount() const noexcept {
+size_t Module::GetEvalModelOutputCount() const noexcept {
   return eval_output_names_.size();
 }
 
+void Module::GetTrainModelOutputName(size_t& index, std::string& name) const {
+  ORT_ENFORCE(index < train_output_names_.size(), "Train output name index out of range");
```

Comment:
May we provide more context? Like `"Train output name index out of range. Expected in range [0-", GetTrainModelOutputCount(), "). Actual ", index, "."`